### ISSUE TITLE
LaTeX equation numbering does not work in Typora 1.7.5

### ISSUE THREAD

**Issue by adk3141** [typora/typora-issues#5855]
Equation numbers fail to appear (and references to them fail) in Typora 1.7.5. The same document work properly in the previous version.

To reproduce: I seems that any example with labelled equations fails, e.g.,
$$
\begin{equation}
  x = y
  \label{foo}
\end{equation}
$$
No label appears, and references such as `$\ref{equation}$` don't work.

 - OS: Windows 10 64bit
   Typora Version: 1.7.5

Markdown preferences do not make any difference. Equations numbers do not appear however the options are set.

**Comment by abnerlee** [comment]
The default behavior is same as https://www.mathjax.org/#demo

To enable math numbering, please enable following option and restart Typora.

<img width="832" alt="Screenshot 2024-05-27 at 10 47 21" src="https://github.com/typora/typora-issues/assets/1621242/d60dff96-e367-48bd-a5ed-0614e23d7fe9">

**Comment by adk3141** [comment]
# LaTeX equation numbering does not work in Typora 1.7.5 **#5855**

If *No Automatic Equation Numbering* is set, any example with labelled equations fails, e.g.,
$$
\begin{equation}
  x = y \label{foo}
\end{equation}
$$
No label appears, and references such as $(\ref{foo})$ doesn't work, although it did prior to Typora 1.7.5. If the *Auto Numbering for All Math Equations* is enabled then this works, but *all* equations such as
$$
\int_{-\infty}^\infty dx\,e^{-\frac12x^2} = \sqrt{2\pi}
$$
are numbered *unless* `\nonumber` is explicitly specified
$$
\int_{-\infty}^\infty dx\,e^{-\frac12x^2} = \sqrt{2\pi} \nonumber,
$$
If *Use AMS numbering rules* is specified then everything behaves as I would expect. Before Typora 1.7.5 this was also the behavior for *No Automatic Equation Numbering*, so it seems that an undocumented change was made.

It is interesting that The *Auto numbering math equations* choice only appears to have an effect after restarting Typora, using *Refresh All Math Expressions* has no effect. Is this intentional?

Note that `$$ ... $$` in TeX or LaTeX does no such implicit numbering, so the behavior has nothing to do with AMSTeX *per se*.
